src/libs/actions/Report.ts

```diff
@@ -2344,6 +2377,58 @@ function getCurrentUserAccountID(): number {
     return currentUserAccountID;
 }
 
+function navigateToMostRecentReport(reportID: string, isChatThread: boolean) {
+    const sortedReportsByLastRead = ReportUtils.sortReportsByLastRead(Object.values(allReports ?? {}) as Report[], reportMetadata);
+
+    // We want to filter out the current report, hidden reports and empty chats
+    const filteredReportsByLastRead = sortedReportsByLastRead.filter(
+        (sortedReport) =>
+            sortedReport?.reportID !== reportID &&
+            sortedReport?.notificationPreference !== CONST.REPORT.NOTIFICATION_PREFERENCE.HIDDEN &&
+            ReportUtils.shouldReportBeInOptionList({
+                report: sortedReport,
+                currentReportId: '',
+                isInGSDMode: false,
+                betas: [],
+                policies: {},
+                excludeEmptyChats: true,
+                doesReportHaveViolations: false,
+                includeSelfDM: true,
+            }),
+    );
+    const lastAccessedReportID = filteredReportsByLastRead.at(-1)?.reportID;
+
+    if (lastAccessedReportID) {
+        // If it is not a chat thread we should call Navigation.goBack to pop the current route first before navigating to last accessed report.
+        if (!isChatThread) {
+            Navigation.goBack();
+        }
+        Navigation.navigate(ROUTES.REPORT_WITH_ID.getRoute(lastAccessedReportID));
+    } else {
+        const participantAccountIDs = PersonalDetailsUtils.getAccountIDsByLogins([CONST.EMAIL.CONCIERGE]);
+        const chat = ReportUtils.getChatByParticipants(participantAccountIDs);
+        if (chat?.reportID) {
+            // If it is not a chat thread we should call Navigation.goBack to pop the current route first before navigating to Concierge.
+            if (!isChatThread) {
+                Navigation.goBack();
+            }
+            Navigation.navigate(ROUTES.REPORT_WITH_ID.getRoute(chat.reportID));
+        }
+    }
+}
+
+function leaveGroupChat(reportID: string) {
+    const optimisticData: OnyxUpdate[] = [
+        {
+            onyxMethod: Onyx.METHOD.SET,
+            key: `${ONYXKEYS.COLLECTION.REPORT}${reportID}`,
+            value: null,
+        },
+    ];
+    navigateToMostRecentReport(reportID, false);
+    API.write(WRITE_COMMANDS.LEAVE_GROUP_CHAT, {reportID}, {optimisticData});
```

Comment:
There's not much that would prevent someone from leaving a Group chat so we went with "fire and forget" in terms of error handling.

I suppose we could put the report data _back_. But that seems undesirable.

I notice that's what we do for `LeaveRoom` [here](https://github.com/Expensify/App/blob/05eb481875faaafa1bcd2077a3c4d10a36a89133/src/libs/actions/Report.ts#L2396-L2402) and I don't really understand why or what would even prevent someone from leaving a room _or_ a Group Chat under normal circumstances.

Reply:
Yeah, there isn't really anything that would prevent someone from leaving a room, assuming they're already a member of it, which is why I went with this approach.